packages/flutter_tools/lib/src/resident_web_runner.dart

```diff
@@ -0,0 +1,165 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'asset.dart';
+import 'base/common.dart';
+import 'base/file_system.dart';
+import 'base/terminal.dart';
+import 'build_info.dart';
+import 'bundle.dart';
+import 'dart/package_map.dart';
+import 'device.dart';
+import 'globals.dart';
+import 'project.dart';
+import 'resident_runner.dart';
+import 'run_hot.dart';
+import 'web/asset_server.dart';
+import 'web/compile.dart';
+import 'web/web_device.dart';
+
+/// A hot-runner which handles browser specific delegation.
+class ResidentWebRunner extends ResidentRunner {
+  ResidentWebRunner(
+    List<FlutterDevice> flutterDevices, {
+    String target,
+    @required this.flutterProject,
+    @required bool ipv6,
+  }) : super(
+          flutterDevices,
+          target: target,
+          usesTerminalUI: true,
+          stayResident: true,
+          saveCompilationTrace: false,
+          debuggingOptions: DebuggingOptions.enabled(
+            const BuildInfo(BuildMode.debug, ''),
+          ),
+          ipv6: ipv6,
+        );
+
+  WebAssetServer _server;
+  ProjectFileInvalidator projectFileInvalidator;
+  DateTime _lastCompiled;
+  final FlutterProject flutterProject;
+
+  @override
+  Future<int> attach(
+      {Completer<DebugConnectionInfo> connectionInfoCompleter,
+      Completer<void> appStartedCompleter}) async {
+    connectionInfoCompleter?.complete(DebugConnectionInfo());
+    setupTerminal();
+    final int result = await waitForAppToFinish();
+    await cleanupAtFinish();
+    return result;
+  }
+
+  @override
+  Future<void> cleanupAfterSignal() {
+    return null;
```

Comment:
Why does the server not need to be cleaned up here?

Reply:
Done